### ISSUE TITLE
Dependencies sometimes partially load, leaving nses as #js {} or nil

### ISSUE THREAD

**Issue by lvh** [bhauman/lein-figwheel#210]
Documented in detail upstream: https://github.com/andrewmcveigh/cljs-time/issues/41#issuecomment-129969406

TL;DR:

When using `cljs-time`, some namespaces are `nil` or `#js {}` when requiring them dynamically, e.g., `(require 'cljs-time.core)` or through nREPL.

I resolved the issue by adding libspecs in my `ns` forms such that cljs-time.core is depended upon through a ns that Fighweel can find when tracking dependencies starting from the main ns. The `cljs-time.core` libspec wasn't in my core ns, but requiring the ns that does require `cljs-time.core` from the core ns allowed Figwheel's dependency resolution system to find the dependency, resolving the issue.

Dixit @bhauman:

> The problem isn't cljs-time its a library resolution problem I think.  It would be cool if you could verify that this isn't a problem in the regular cljs.repl

FWIW, the problem does occur with lein figwheel's repl, but not with lein cljsbuild repl.

**Comment by bhauman** [comment]
Just letting you know that I am working on this now. :)
